float32 fXPos
float32 fYPos
float32 fYawAngle
